### PR TITLE
CSPROJ & Sample Projects Spring Cleaning

### PR DESCRIPTION
**Pull request by jmacato**
- What does the pull request do?
  * Cleaned up all csproj files that can be cleaned.
  * Ported BindingTest, RenderTest and VirtualizationTest to the newer csproj format.
  * Added a common props file `SharedVersion.props` for repo-wide package and versioning information

### CODE REVIEW

**Comment by jmacato** [comment]
that makes more sense sir, i was aware of it but i didn't totally understood how it worked

File: src/Shared/Shared.props [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<Project DefaultTargets="Build"
```
[the remark this answers is not in the document]
Comment (jmacato) [reply src/Shared/Shared.props]: that makes more sense sir, i was aware of it but i didn't totally understood how it worked

**Review by grokys — CHANGES_REQUESTED** [review]
Hmm, this seems to have broken BindingTest/RenderTest/VirtualizationTest on net 461. I get the following error on running:

```
System.InvalidOperationException: 'No windowing subsystem found. Are you missing assembly references?'
```

I think this is because the Win32 and Direct2D1 libraries need to be referenced under .net fx, as it tries to determine the windowing/rendering subsystems at runtime depending on which libraries are loaded:

https://github.com/AvaloniaUI/Avalonia/blob/ef3afccaa87248dfd38d5e0cc9a83bf36aadfca0/src/Avalonia.DotNetFrameworkRuntime/AppBuilder.cs#L58

**Comment by grokys** [comment]
Adding the following to the .csproj seems to fix it:

```
  <ItemGroup Condition="'$(TargetFramework)'=='net461'">
    <ProjectReference Include="..\..\src\Windows\Avalonia.Win32\Avalonia.Win32.csproj" />
    <ProjectReference Include="..\..\src\Windows\Avalonia.Direct2D1\Avalonia.Direct2D1.csproj" />
  </ItemGroup>
```

and 

```
  <Import Condition="'$(TargetFramework)'=='net461'" Project="..\..\build\SharpDX.props" />
```

However if I do that I get the error:

```
1>C:\Program Files\dotnet\sdk\2.2.0-preview1-007622\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(114,5): error : Cannot find project info for 'D:\projects\AvaloniaUI\Avalonia\src\Windows\Avalonia.Win32\Avalonia.Win32.csproj'. This can indicate a missing project reference.
```

Which seems to be https://github.com/dotnet/sdk/issues/1227

**Comment by grokys** [comment]
@kekekeks @jkoritzinsky should we still have the difference in the way .net fx and .net core decide which libraries to use? Can't they both use the same logic on both platforms so we don't have to do the conditional project references?

**Review by grokys — APPROVED** [review]
LGTM. I notice that Avalonia.Markup.Xaml hasn't been converted but that's a bit more complicated due to the inclusion of the submodule. Lets do that one separately if possible.

**Comment by grokys** [comment]
@kekekeks we have a designer test failing on CI:

```
System.TypeLoadException : Unable to find any of types: Avalonia.DesignerSupport.DesignerAssist
      Stack Trace:
        tests\Avalonia.DesignerSupport.Tests\Checker.cs(71,0): at Avalonia.DesignerSupport.Tests.Checker.LookupType(String[] names)
        tests\Avalonia.DesignerSupport.Tests\Checker.cs(76,0): at Avalonia.DesignerSupport.Tests.Checker.LookupStaticMethod(String typeName, String method)
        tests\Avalonia.DesignerSupport.Tests\Checker.cs(30,0): at Avalonia.DesignerSupport.Tests.Checker.DoCheck(String baseAsset, String xamlText)
           at Avalonia.DesignerSupport.Tests.Checker.DoCheck(String baseAsset, String xamlText)
        tests\Avalonia.DesignerSupport.Tests\DesignerSupportTests.cs(29,0): at Avalonia.DesignerSupport.Tests.DesignerSupportTests.DesgignerApiShoudBeOperational(String outputDir, String xamlFile)
```

However this passes locally on my machine. Any ideas?

**Comment by grokys** [comment]
Ok, I've disabled that test for now. Going to get this merged.

**Comment by jmacato** [comment]
@grokys thank you sir!

**Comment by kekekeks** [comment]
This test should be updated to use the new client-server previewer infrastructure anyway, so it's probably fine.

**Pull request merged**